buffer/src/main/java/io/netty/buffer/api/ByteCursor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.buffer.api;
+
+import io.netty.util.ByteProcessor;
+
+/**
+ * The ByteCursor scans through a sequence of bytes.
+ * This is similar to {@link ByteProcessor}, but for external iteration rather than internal iteration.
+ * The external iteration allows the callers to control the pace of the iteration.
+ * The API includes methods for reading {@code long}s as a batch of 8 bytes.
+ * The long values are always in big-endian format, so that the highest-order byte in the long value, contain the byte
+ * that would otherwise have been returned by the next call to {@link #getByte()}.
+ */
+public interface ByteCursor {
+    /**
+     * Check if the iterator has at least 8 bytes left, and if so, read those 8 bytes and move the cursor forward.
+     * The bytes are packed as a {@code long} value in big-endian format, such that the highest-order byte
```

Comment:
Ping @chrisvest

Reply:
You are thinking of this as a way of accessing the buffer in units, rather than as a stream of bytes. If we are streaming bytes, and grab batches of 8 at a time as a long, if that long was in little-endian format it would have the bytes in reverse. If you want to access units, why not use the read or get methods?

Is this API perhaps too hard to understand?

Reply:
Removing the batch iteration from ByteCursor will make some operations slower, but we can do that.